### PR TITLE
[docs]: Document environment variables

### PR DESCRIPTION
**Pull request by stavros-k**
This is a very early draft, info might be incorrect at the moment.
Just gone through code with search and picked what I could find.

Feel free to leave any comments for improvements, I'll continue to poke around to find more.

---

Formatting isn't checked yet, so expect things to look weird at the moment

---

Removed some `mapbox` references, there are some in the `server/package-lock.json` and  `mobile/ios/Info.plist` which I'm not sure what to do

### CODE REVIEW

File: docs/docs/install/environment-variables.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+| :---------------------------- | :------------------------------------------- | :-----------------------------------------: | :-------------------------------------- |
+| `TZ`                          | Timezone                                     |                                             | microservices                           |
+| `NODE_ENV`                    | Environment (production, development)        |                `production`                 | server, microservices, machine learning |
+| `LOG_LEVEL`                   | Log Level (verbose, debug, log, warn, error) |                    `log`                    | server, microservices                   |
+| `DISABLE_REVERSE_GEOCODING`   | Disable Reverse Geocoding Precision          |                   `false`                   | microservices                           |
+| `REVERSE_GEOCODING_PRECISION` | Reverse Geocoding Precision                  |                     `3`                     | microservices                           |
+| `PUBLIC_LOGIN_PAGE_MESSAGE`   | Public Login Page Message                    | `My Family Photos and Videos Backup Server` | web                                     |
```
Comment (bo0tzz) [inline docs/docs/install/environment-variables.md:12]: I believe the default for `PUBLIC_LOGIN_PAGE_MESSAGE` is an empty string

File: docs/docs/install/environment-variables.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+| `REDIS_HOST`     | Redis Host     | `immich_redis` | server, microservices |
+| `REDIS_PORT`     | Redis Port     |     `6379`     | server, microservices |
+| `REDIS_INDEX`    | Redis Index    |      `0`       | server, microservices |
+| `REDIS_USERNAME` | Redis Username |                | server, microservices |
+| `REDIS_PASSWORD` | Redis Password |                | server, microservices |
+| `REDIS_SOCKET`   | Redis Socket   |                | server, microservices |
+
+:::info
+
+`REDIS_URL` must start with `ioredis://` and a `base64` encoded JSON string.
```
Comment (bo0tzz) [inline docs/docs/install/environment-variables.md:85]: Maybe include a link to the ioredis docs?

**Comment by stavros-k** [comment]
I think all env's I could find are added.

Please take a look and let me know if something is missing

File: docs/docs/install/environment-variables.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+| :---------------------------- | :------------------------------------------- | :-----------------------------------------: | :------------------------------------------- |
+| `TZ`                          | Timezone                                     |                                             | microservices                                |
+| `NODE_ENV`                    | Environment (production, development)        |                `production`                 | server, microservices, machine learning, web |
+| `LOG_LEVEL`                   | Log Level (verbose, debug, log, warn, error) |                    `log`                    | server, microservices                        |
+| `DISABLE_REVERSE_GEOCODING`   | Disable Reverse Geocoding Precision          |                   `false`                   | microservices                                |
+| `REVERSE_GEOCODING_PRECISION` | Reverse Geocoding Precision                  |                     `3`                     | microservices                                |
```
Comment (jrasm91) [inline docs/docs/install/environment-variables.md:11]: Can you put reverse geocoding related env variables under their own section?

File: docs/docs/install/environment-variables.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+## Paths
+
+| Variable                           | Description                      |           Default            | Services              |
+| :--------------------------------- | :------------------------------- | :--------------------------: | :-------------------- |
+| `IMMICH_MEDIA_LOCATION`            | Media Location                   |          `./upload`          | server, microservices |
+| `MACHINE_LEARNING_CACHE_FOLDER`    | ML Cache Location                |           `/cache`           | machine learning      |
+| `TRANSFORMERS_CACHE`               | ML Transformers Cache Location   |           `/cache`           | machine learning      |
+| `REVERSE_GEOCODING_DUMP_DIRECTORY` | Reverse Geocoding Dump Directory | `./.reverse-geocoding-dump/` | microservices         |
+| `TYPESENSE_DATA_DIR`               | Typesense Data Directory         |           `/data`            | typesense             |
```
Comment (jrasm91) [inline docs/docs/install/environment-variables.md:49]: I would rather move media location to general, ML cache folder and transformers cache to ML section, geocoding to it's own section, and typesense data dir to the typesense section.

File: docs/docs/install/environment-variables.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+| `REDIS_HOSTNAME` | Redis Host     | `immich_redis` | server, microservices |
+| `REDIS_PORT`     | Redis Port     |     `6379`     | server, microservices |
+| `REDIS_DBINDEX`  | Redis DB Index |      `0`       | server, microservices |
+| `REDIS_USERNAME` | Redis Username |                | server, microservices |
+| `REDIS_PASSWORD` | Redis Password |                | server, microservices |
+| `REDIS_SOCKET`   | Redis Socket   |                | server, microservices |
+
+:::info
+
+`REDIS_URL` must start with `ioredis://` and a `base64` encoded JSON string.
```
Comment (jrasm91) [inline docs/docs/install/environment-variables.md:88]:
```suggestion
`REDIS_URL` must start with `ioredis://` and then include a `base64` encoded JSON string for the configuration.
```

File: docs/docs/install/environment-variables.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+| `TYPESENSE_ENABLED`  | Enable Typesense   |             | server, microservices            |
+| `TYPESENSE_URL`      | Typesense URL      |             | server, microservices            |
+| `TYPESENSE_HOST`     | Typesense Host     | `typesense` | server, microservices            |
+| `TYPESENSE_PORT`     | Typesense Port     |   `8108`    | server, microservices            |
+| `TYPESENSE_PROTOCOL` | Typesense Protocol |   `http`    | server, microservices            |
+| `TYPESENSE_API_KEY`  | Typesense API Key  |             | server, microservices, typesense |
+
+:::info
+
+`TYPESENSE_URL` must start with `ha://` and a `base64` encoded JSON string.
```
Comment (jrasm91) [inline docs/docs/install/environment-variables.md:131]:
```suggestion
`TYPESENSE_URL` must start with `ha://` and then include a `base64` encoded JSON string for the configuration.
```

File: docker/example.env [hunk 7]
```diff
@@ -52,10 +52,10 @@ TYPESENSE_API_KEY=some-random-text
...
 # [
-#      { 'host': 'typesense-1.example.net', 'port': '443', 'protocol': 'https' },
-#      { 'host': 'typesense-2.example.net', 'port': '443', 'protocol': 'https' },
-#      { 'host': 'typesense-3.example.net', 'port': '443', 'protocol': 'https' },
-#  ]
+#      { "host": "typesense-1.example.net", "port": "443", "protocol": "https" },
+#      { "host": "typesense-2.example.net", "port": "443", "protocol": "https" },
+#      { "host": "typesense-3.example.net", "port": "443", "protocol": "https" },
+# ]
 # TYPESENSE_URL=ha://WwogICAgeyAnaG9zdCc6ICd0eXBlc2Vuc2UtMS5leGFtcGxlLm5ldCcsICdwb3J0JzogJzQ0MycsICdwcm90b2NvbCc6ICdodHRwcycgfSwKICAgIHsgJ2hvc3QnOiAndHlwZXNlbnNlLTIuZXhhbXBsZS5uZXQnLCAncG9ydCc6ICc0NDMnLCAncHJvdG9jb2wnOiAnaHR0cHMnIH0sCiAgICB7ICdob3N0JzogJ3R5cGVzZW5zZS0zLmV4YW1wbGUubmV0JywgJ3BvcnQnOiAnNDQzJywgJ3Byb3RvY29sJzogJ2h0dHBzJyB9LApd
```
Comment (jrasm91) [inline docker/example.env:59]:
```suggestion
# TYPESENSE_URL=ha://WwogIHsgImhvc3QiOiAidHlwZXNlbnNlLTEuZXhhbXBsZS5uZXQiLCAicG9ydCI6ICI0NDMiLCAicHJvdG9jb2wiOiAiaHR0cHMiIH0sCiAgeyAiaG9zdCI6ICJ0eXBlc2Vuc2UtMi5leGFtcGxlLm5ldCIsICJwb3J0IjogIjQ0MyIsICJwcm90b2NvbCI6ICJodHRwcyIgfSwKICB7ICJob3N0IjogInR5cGVzZW5zZS0zLmV4YW1wbGUubmV0IiwgInBvcnQiOiAiNDQzIiwgInByb3RvY29sIjogImh0dHBzIiB9Cl0=
```

**Review by jrasm91 — COMMENTED** [review]
Looks good. There is another class of environment variables that I think we should also include (at the top probably). They are not used in the containers directly, but indirectly via the docker-compose file.

Specifically, 
- `IMMICH_VERSION` (tag for the images, defaults to release)
- `UPLOAD_LOCATION` (helper for the upload volume mount, required)

**Review by jrasm91 — APPROVED** [review]

**Comment by bo0tzz** [comment]
@stavros-k thank you for this, much appreciated!

**Pull request merged**